### PR TITLE
feat: add waitForSubgraphSYnc (#109)

### PR DESCRIPTION
**Pull request by aouahib**

### CODE REVIEW

**Comment by aouahib** [comment]
Current dependencies on/for this PR:
* main
  * **PR #153** <a href="https://app.graphite.dev/github/pr/metastreet-labs/metastreet-sdk/153" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/metastreet-labs/metastreet-sdk/153?utm_source=stack-comment).

File: apps/test/src/components/PositionsSection.tsx
Comment (robence) [inline apps/test/src/components/PositionsSection.tsx:130]: Did you mean a finally block?

**Review by robence — COMMENTED** [review]
Seems good, added one comment for loading state

File: apps/test/src/components/PositionsSection.tsx
Comment (aouahib) [reply apps/test/src/components/PositionsSection.tsx:136]: not really, why?

Comment (robence) [reply apps/test/src/components/PositionsSection.tsx:136]: I see, this is a void function. In this case there's no benefit 👍 
If it were to change you'd have to look out for it though, a return statement in the try block would cause the setLoading to not run.

**Review by robence — APPROVED** [review]

**Review by andrewszucs — APPROVED** [review]

**Comment by aouahib** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/metastreet-labs/metastreet-sdk/153?utm_source=stack-comment).


Job Reference: LhE62Pbrj0P1k6gpAemc

**Pull request merged**
